frontend/app/lib/api.ts

```diff
@@ -1,28 +1,37 @@
-import { apiUrl } from "../utils/env";
+import { apiUrl } from "../utils/env"
 
 async function fetchWithAuth(endpoint: string, options: RequestInit = {}) {
-  const headers = new Headers(options.headers);
+  if (!apiUrl) {
+    throw new Error("API URL is not configured")
+  }
 
-  headers.set("Content-Type", "application/json");
-  headers.set("Accept", "application/json");
+  const headers = new Headers(options.headers)
+  headers.set("Content-Type", "application/json")
+  headers.set("Accept", "application/json")
 
-  const response = await fetch(`${apiUrl}${endpoint}`, {
-    ...options,
-    headers,
-  });
+  const url = `${apiUrl}${endpoint}`
 
-  if (!response.ok) {
-    const error = await response.json().catch(() => ({}));
-    throw new Error(error.error || "An error occurred");
-  }
+  try {
+    const response = await fetch(url, {
+      ...options,
+      headers,
+    })
+
+    if (!response.ok) {
+      const error = await response.json().catch(() => ({}))
+      throw new Error(error.error || `HTTP error! status: ${response.status}`)
+    }
 
-  return response.json();
+    return response.json()
+  } catch (error) {
+    throw error
```

Comment:
We can even add more context into the rethrown error, or no need to just catch and rethrow as is